File: Data Structures & Algorithms/single-number/submission-6.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        for (int i = 0; i < nums.Length; i++) {
            bool flag = true;
            for (int j = 0; j < nums.Length; j++) {
                if (i != j && nums[i] == nums[j]) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                return nums[i];
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/binary-search/submission-6.java
public class Solution {
    public int binary_search(int l, int r, List<Integer> nums, int target) {
        if (l > r) return -1;
        int m = l + (r - l) / 2;
        
        if (nums.get(m) == target) return m;
        return (nums.get(m) < target) ? 
            binary_search(m + 1, r, nums, target) : 
            binary_search(l, m - 1, nums, target);
    }

    public int search(List<Integer> nums, int target) {
        return binary_search(0, nums.size() - 1, nums, target);
    }
}